utils/dependency_management.py

```diff
@@ -294,10 +294,20 @@ def check_requirements(extras, github_annotate):
 
         for requirement_abstract in requirements_abstract:
             for requirement_concrete in requirements_concrete:
-                version = Specifier(str(requirement_concrete.specifier)).version
-                if canonicalize_name(requirement_abstract.name) == canonicalize_name(
-                    requirement_concrete.name
-                ) and requirement_abstract.specifier.contains(version):
+                if '@' in str(requirement_concrete):
+                    # `@` is not listed as a valid `specifier` in `class Specifier`.
```

Comment:
```suggestion
                    # For versions that are use `@` specifier as for git repos
                    # since `@` is not listed as a valid `specifier` in class `Specifier`.
```